src/GraphQlClientGenerator/BaseClasses.cs

```diff
@@ -177,7 +177,11 @@ public static string BuildArgumentValue(object value, string formatMask, Formatt
         }
 
         if (value is JObject jObject)
-            return BuildEnumerableArgument(jObject, null, formatting, level + 1, indentationSize, '{', '}');
+        {
+            var str = jObject.ToString(formatting == Formatting.Indented ? Newtonsoft.Json.Formatting.Indented : Newtonsoft.Json.Formatting.None);
+            var escapedStr = Newtonsoft.Json.JsonConvert.ToString(str);
+            return escapedStr;
+        }
```

Comment:
We can't propose this change as PR into the original repo, it's very bespoke. This type serialization if/else should be refactored to allow custom implementations injection per type